library/ZendService/Twitter/Response.php

```diff
@@ -51,21 +57,23 @@ class Response
      * @param  HttpResponse $httpResponse
      * @throws Exception\DomainException if unable to decode JSON response
      */
-    public function __construct(HttpResponse $httpResponse)
+    public function __construct(HttpResponse $httpResponse = null)
     {
         $this->httpResponse = $httpResponse;
-        $this->rawBody      = $httpResponse->getBody();
-        try {
-            $jsonBody = Json::decode($this->rawBody, Json::TYPE_OBJECT);
-            $this->jsonBody = $jsonBody;
-        } catch (JsonException $e) {
-            throw new Exception\DomainException(sprintf(
-                'Unable to decode response from twitter: %s',
-                $e->getMessage()
-            ), 0, $e);
+
+        if (! is_null($httpResponse) and ! empty($httpResponse->getBody())) {
+            $this->populate($httpResponse);
         }
     }
 
+    protected function setRateLimit($headers = [])
+    {
+        $this->rateLimit['limit'] = isset($headers['x-rate-limit-limit']) ? $headers['x-rate-limit-limit'] : 0;
```

Comment:
Please use a single assignment here: `$this->rateLimit = [ 'key1' => expression, 'key2' => expression]`. Adding keys programmatically makes the code harder to understand, as it could lead to the assumption that different keys are set